sktime/utils/_testing/deep_equals.py

```diff
@@ -118,7 +118,7 @@ def ret(is_equal, msg):
         )
     elif type(x).__name__ == "ForecastingHorizon":
         return ret(*_fh_equals(x, y, return_msg=True))
-    elif x != y:
+    elif np.any(x != y):
```

Comment:
this is going to break things!

I would do instead:
1. compute `x != y`
2. check if boolean. if yes, return that
3. else, return `np.any` . (perhaps check first if that makes sense, and if not, you could raise an exception)

Reply:
I changed it as you proposed.

Checking if `np.any` makes sense difficult since `np.any` allows anything as input which is array_like. This does not mean that it will work with all array_like objects. E.g., `np.any(np.array("string"))` will fail with `TypeError: cannot perform reduce with flexible type`. Honestly, I do not know how to test it. 
However, as long as `__ne__` returns bool or a list of bools it should work.

Reply:
ok, this works, I suppose.